rq/plan.sh

```diff
@@ -27,5 +26,31 @@ do_build() {
 
 do_install() {
   mkdir -pv "$pkg_prefix/bin"
-  install -v "$HAB_CACHE_SRC_PATH/$pkg_filename" "$pkg_prefix/bin/$pkg_name"
+  install -v "rq" "$pkg_prefix/bin/$pkg_name"
+
+  local glibc_libs=(
+    libdl.so.2
+    librt.so.1
+    libpthread.so.0
+    libc.so.6
+    ld-linux-x86-64.so.2
+    libm.so.6
+  )
+  for lib in "${glibc_libs[@]}"; do
+    build_line "Patching ELF for ${lib}"
+    patchelf --interpreter "$(pkg_path_for core/glibc)/lib/${lib}" --set-rpath "${LD_RUN_PATH}" "${pkg_prefix}/bin/rq"
```

Comment:
I don't think this is correct. The `--interpreter` flag is setting the INTERP program header.  So in this code, you end up setting the interpreter to libgcc_s.so.1 since that is the last call you make down below.

Unfortunately, since that isn't an executable file you'll get:

```
[46][default:/src:0]# /hab/pkgs/core/rq/0.10.4/20171205110032/bin/rq
bash: /hab/pkgs/core/rq/0.10.4/20171205110032/bin/rq: Permission denied
```

If you mark libgcc_s as an executable and run it, you'll get a segmentation fault since libgcc_s.so doesn't know how to load the program and presumably starts doing something very wrong.

You should be able to replace lines 31-51 with:

```
patchelf --interpreter "$(pkg_path_for)/lib/ld-linux-x86-64.so.2" --set-rpath "${LD_RUN_PATH}" "${pkg_prefix}/bin/rq"
```

The dynamic loader should resolve any needed libraries via the RPATH.

Reply:
That's hugely better, yes. I only pretend to know what I'm doing with elves and patches.